internal/gengapic/genrest.go

```diff
@@ -245,36 +306,40 @@ func (g *generator) generateQueryString(m *descriptor.MethodDescriptorProto) {
 	}
 	sort.Strings(fields)
 
-	p("queryParamStrs := []string{}")
+	requestObject := "req"
+
+	p("params := url.Values{}")
 	// TODO(dovs): make this work for nested fields
-	for _, f := range fields {
-		field := queryParams[f]
-		f = snakeToCamel(f)
+	for _, path := range fields {
+		field := queryParams[path]
+		accessor := buildAccessor(path)
+		closer := "}"
 		if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REPEATED {
 			// It's a slice, so check for nil
-			p("if req.Get%s() != nil {", f)
-		} else {
+			p("if %s%s != nil {", requestObject, accessor)
+		} else if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_OPTIONAL {
 			// Default values are type specific
 			switch field.GetType() {
+			// Degenerate case, field should never be a message because that implies it's not a leaf.
 			case descriptor.FieldDescriptorProto_TYPE_MESSAGE, descriptor.FieldDescriptorProto_TYPE_BYTES:
-				p("if req.Get%s() != nil {", f)
+				p("if %s%s != nil {", requestObject, accessor)
 			case descriptor.FieldDescriptorProto_TYPE_STRING:
-				p(`if req.Get%s() != "" {`, f)
+				p(`if %s%s != "" {`, requestObject, accessor)
 			case descriptor.FieldDescriptorProto_TYPE_BOOL:
-				p(`if req.Get%s() {`, f)
-			default: // Handles all numeric types
-				p(`if req.Get%s() != 0 {`, f)
+				p(`if %s%s {`, requestObject, accessor)
+			default: // Handles all numeric types including enums
+				p(`if %s%s != 0 {`, requestObject, accessor)
 			}
+		} else {
+			// This is a required field, so we must include the field
```

Comment:
If the field _isn't_ labeled with proto3_optional, that doesn't mean its _required_ does it? I guess our best indicator is `option (google.api.field_behavior) = REQUIRED`?

Reply:
You're right, I misread the enum and thought that `OPTIONAL` was the default enum value, when `0` is explicitly reserved for errors.